drivers/DeviceKey.h

```diff
@@ -0,0 +1,148 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICEKEY_H
+#define MBED_DEVICEKEY_H
+
+#include "stddef.h"
+#include "stdint.h"
+#include "platform/NonCopyable.h"
+
+namespace mbed {
+/** \addtogroup drivers */
+
+#define DEVICE_KEY_16BYTE 16
+#define DEVICE_KEY_32BYTE 32
+
+enum DeviceKeyStatus {
```

Comment:
These are return types: 
- why then methods have `int` return type ?
- not part of the class?

Reply:
Using the enum can cause type problems in C++. Especially when errors need to be propogated upwards. Just using int simplifies a lot of things for users.

Either way @SenRamakri has bigger plans with error codes.

Reply:
@0xc0170 @geky 
There is a process now to add Error Codes to mbed-os and when this happen, we will change the implementation to use the new error codes mechanism. 
Until then We thought it is better for our code to use an internal error code mechanism and return int in order to excuse the caller from the need of using our enum.

Reply:
Sounds good to me. Still does not answer my second question why is this nor part of the class itself.

Reply:
@0xc0170 
Actually only because it mimics the idea of the future solution of _MbedErrorStatus enum. I don't mind to make it part of the class but it’s not the way it would be implemented in the future.

Reply:
Personally in these situations I still use the enum but then use a typdef in any location where data might be used by another processor.

Worst case I'd rather a typedef be used than a bare int.